src/Osma.Mobile.App/ViewModels/CreateInvitation/CreateInvitationViewModel.cs

```diff
@@ -0,0 +1,94 @@
+﻿using System;
+using System.Reactive.Linq;
+using System.Threading.Tasks;
+using System.Windows.Input;
+using Acr.UserDialogs;
+using AgentFramework.Core.Contracts;
+using Osma.Mobile.App.Services.Interfaces;
+using Osma.Mobile.App.ViewModels.Account;
+using ReactiveUI;
+using Xamarin.Forms;
+using ZXing.Net.Mobile.Forms;
+using AgentFramework.Core.Extensions;
+
+namespace Osma.Mobile.App.ViewModels.CreateInvitation
+{
+    public class CreateInvitationViewModel : ABaseViewModel
+    {
+        private readonly ICustomAgentContextProvider _agentContextProvider;
+        private readonly IConnectionService _connectionService;
+
+        public CreateInvitationViewModel(
+            IUserDialogs userDialogs,
+            INavigationService navigationService,
+            ICustomAgentContextProvider agentContextProvider,
+            IConnectionService defaultConnectionService
+            ) : base(
+                "CreateInvitation",
+                userDialogs,
+                navigationService
+           )
+        {
+            _agentContextProvider = agentContextProvider;
+            _connectionService = defaultConnectionService;
+        }
+
+        public override async Task InitializeAsync(object navigationData)
+        {
+            await base.InitializeAsync(navigationData);
+        }
+
+        private async Task CreateInvitation()
+        {
+            try
+            {
+                var context = await _agentContextProvider.GetContextAsync();
+                var (invitation, _) = await _connectionService.CreateInvitationAsync(context);
+
+                string barcodeValue = invitation.ServiceEndpoint + "?c_i=" + (invitation.ToJson().ToBase64());
```

Comment:
You can use the following util from agent framework here https://github.com/streetcred-id/agent-framework/blob/master/src/AgentFramework.Core/Utils/MessageUtils.cs#L29

Reply:
@tplooker But it will encode it in this format 
$"{baseUrl}?m={message.ToJson().ToBase64()}

we need c_i as the query parameter. Should I then enrich the agent framework method first to take a param to understand which query param to use?

Reply:
Correct the url format is yet to change in aries, I am happy to accept this PR in the meantime.

Reply:
Currently we have the following code in the CreateInvitationViewModel.cs

string barcodeValue = invitation.ServiceEndpoint + "?c_i=" + MessageUtils(invitation.ToJson().ToBase64()); 

This code does not compile as there is no invocable constructor for MessageUtils. This is causing the compile to fail.